contracts/src/extensions/briding/portals/KeystoreBasePortal.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.27;
+
+import {ICrossDomainMessenger} from "optimism-interfaces/universal/ICrossDomainMessenger.sol";
+
+import {L2ToL1MsgSenderIsNotThisContract, L2ToL1TxSenderIsNotRollupContract} from "./PortalErrors.sol";
+import {ReceiverAliasedAddress} from "./ReceiverAliasedAddress.sol";
+
+contract KeystoreBasePortal is ReceiverAliasedAddress {
```

Comment:
can you just have an opstack version of this contract that accepts a `L1_CROSS_DOMAIN_MESSENGER` in the constructor and stores it in an `immutable` variable? then just create child contracts that simply pass the relevant address for Base / Optimism / other opstack chains

Reply:
+1. Another option is to make `KeystoreBridge` contracts specific to a master chain. The L1 cross domain messenger is only needed for withdrawals from the master chain to L1, so `KeystoreBridge` could be responsible for configuring withdrawals instead of the portals configuring themselves. Base Wallet, for instance, never needs the L1 cross domain messenger for Optimism or Arbitrum because only deposits will happen there.